internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -74,7 +74,7 @@ func (f *FirstSchedulableWithinLimit) FindCluster(kafka *dbapi.KafkaRequest) (*a
 	//#1
 	clusterObj, err := f.ClusterService.FindAllClusters(criteria)
 	if err != nil {
-		return nil, err
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to find all clusters with criteria: %v", criteria)
```

Comment:
Hey Jameel, there are a few errors changed back such as:
```diff
-		return nil, err
+		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to find all clusters with criteria: %v", criteria)
```

and then back [in another commit](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1201/commits/3a6a8673a46df5fb6275b2ca8cf6bfed708b6d1e#diff-5ffcdb3a533058503533bac73799b825a292e9cc5fabc494e6ec10d4d26753f6L77-R76) to:
```diff
-		return nil, errors.NewWithCause(errors.ErrorGeneral, err, "failed to find all clusters with criteria: %v", criteria)
+		return nil, err
```

I was just wondering what the reasoning for this was?

Reply:
This is to ensure that each commit to refactor each method is isolated. I didn't want to change the return type of a method that I wasn't refactoring. In a later commit, I then refactored the method that calls it to use `error` type instead of a `serviceError` so I no longer had to call `errors.NewWithCause`. 

Isolating each change to a single commit ensures that if we decide to remove one change, we can drop that commit without affecting others that much.

Reply:
Thanks for the explanation, Very nice tip :)